scripts/prune_lm_dir.py

```diff
@@ -328,6 +351,153 @@ def FinalizeOutput(final_work_out):
         os.remove(args.lm_dir_out + "/num_splits")
 
 
+def MatchTargetSize(num_ngrams):
+    assert(args.target_num_ngrams > 0)
+    return abs(num_ngrams - args.target_num_ngrams) / float(args.target_num_ngrams) < args.tolerance
+
+def IterateOnce(scale, step, n_iter, recovery_step, recovery_size):
+    """Prune with a specific threshold and check whether the resulting LM matches the target-num-ngrams
+
+    Args:
+        scale: the threshold equals to scale * args.threshold
+        step: number of current step
```

Comment:
Say:
`      step: the current step-index (each individual operation, such as pruning or E-M, counts as one step).`
     and below [rename the variable to 'iter']
`   'iter: index of current iteration.  We only increase the iteration index when we do an actual pruning
      step, so it will be <= the step index.'`
.. or something like that.